ruby_02-web_applications_with_ruby/outlines/rack_test_in_sinatra.markdown

```diff
@@ -9,200 +9,66 @@ tags: testing, tdd, sinatra, http, rack
-genre.errors[:title]
-```
-
-Let's try it out in tux. 
-
-## Key Points and Other Things
-
-* Use the `rack-test` gem to test rack applications (e.g. Sinatra, Rails)
-* Rack-test hooks in at the level of Rack, so it calls your code the same as a real web request, but with mock objects
-* Declare an `app` method so it knows what Rack app to use
-
-* What can you do with Rack::Test? 
-  * get access to the methods by including `include Rack::Test::Methods`
-  * make a request: (`get/post/put/patch/delete`)
-  * pass params by providing a params hash as the second argument: `post '/', { title: "My Idea", description: "it's hard to come up with descriptions" }`
-  * follow a redirect: `follow_redirect!`
-  * get the request or response: `last_request` or `last_response`
-
-* What to use `last_response` for:
-  * `last_response.status` codes: (200, 404, 302, 500, etc) -- see all status codes [here](http://www.w3.org/Protocols/rfc2616/rfc2616-sec10.html)
-  * `last_response.body`
-
-* Make assertions about the body
-  * look for relevant strings using normal string methods like `.include?`, etc.
-
-## Resources
-
-* [Testing Sinatra with Rack::Test](http://www.sinatrarb.com/testing.html)
-* [Rack Test Docs](https://github.com/brynary/rack-test/)
+## Setup (before lesson)
+
+* Clone the repo [here](https://github.com/s-espinosa/rack_test_in_sinatra)
+
+## Review: Unit and Feature Tests (Slides - 5 mins)
+
+* Unit tests - our specific function works the way we expect with given inputs
+    * Action: method calls on an object
+    * Result: return values/object state
+    * Test: example test
+* Feature tests with Capybara - user clicks a link, etc.
+    * Action: User interaction with a webpage
+    * Result: New web page with content (HTML)
+    * Test: example test
+* Rush hour
+    * Not sending information in forms, not data that’s entered by a human on a web page
+    * Modeling something closer to a machine to machine interaction
+    * Still need to provide certain responses so that client machines know that they have successfully submitted a request even though we’re not rendering a view.
+    * Need to test this functionality
+        * Action: Submit curl request
+        * Result: HTTP response/Updates to DB
+        * Test: ?
+
+## Lecture (Slides - 10 mins)
+
+* How do we test this?
+    * Rack::Test (via rack-test gem)
+    * Gives us a way to test our controllers
+    * Mimics a request/response cycle
+    * Doesn’t require a view to be rendered for either submissions  or responses (no Capybara)
+* What does rack-test give us?
+    * HTML verbs in our tests (get, post, put, patch, delete)
+    * Ability to pass params using a params hash as second argument (post ‘/‘, {title: “My Idea”, description: “fake descriptions are difficult”}
+    * last_response.status: check status codes (200, 404, 302, 500, etc)
+    * last_response.body: check body content - look for relevant strings using string methods like .include?, etc.
```

Comment:
same as above, let's put all related code chunks in backtick's so it's formatted nicer!